pallets/aura-ext/src/consensus_hook.rs

```diff
@@ -19,28 +19,42 @@
 //!
 //! The velocity `V` refers to the rate of block processing by the relay chain.
 
-use super::pallet;
+use super::{pallet, Aura};
 use cumulus_pallet_parachain_system::{
 	consensus_hook::{ConsensusHook, UnincludedSegmentCapacity},
 	relay_state_snapshot::RelayChainStateProof,
 };
 use frame_support::pallet_prelude::*;
+use sp_consensus_aura::{Slot, SlotDuration};
 use sp_std::{marker::PhantomData, num::NonZeroU32};
 
+const RELAY_CHAIN_MILLIS_PER_BLOCK: u64 = 6000;
+
 /// A consensus hook for a fixed block processing velocity and unincluded segment capacity.
 pub struct FixedVelocityConsensusHook<T, const V: u32, const C: u32>(PhantomData<T>);
 
 impl<T: pallet::Config, const V: u32, const C: u32> ConsensusHook
 	for FixedVelocityConsensusHook<T, V, C>
+where
+	<T as pallet_timestamp::Config>::Moment: Into<u64>,
```

Comment:
`Aura::slot_duration` returns `T::Moment` with a definition
```rust
type Moment: Parameter
			+ Default
			+ AtLeast32Bit // From<u16> + From<u32>
			+ Scale<Self::BlockNumber, Output = Self::Moment>
			+ Copy
			+ MaxEncodedLen
			+ scale_info::StaticTypeInfo;
```
And `SlotDuration` cannot be built out of it.

Reply:
Couldn't come up with any other way. Ideally, I would expect `Aura::slot_duration` to return `SlotDuration`.
As you can see from the type definition, it's not convertible to any numeric primitive, so we have to enforce it in boundary.

Reply:
It's reasonable to enforce it in the boundary as long as that aligns with the instantiations that are typically used in runtime declarations. Seems like a good solution to me.